test/module/dy.spec.js

```diff
@@ -0,0 +1,223 @@
+  });
+
+  it('Testing update', async () => {
+    expect(await model.upsert(item)).to.deep.equal({ created: true });
+    item.age = 55;
+    const result = await model.update(item);
+    expect(result).to.deep.equal(item);
+  });
+
+  it('Testing update with conditions', async () => {
+    expect(await model.upsert(item)).to.deep.equal({ created: true });
+    item.age = 55;
+    const result = await model.update(item, { conditions: { attr: 'age', eq: 50 } });
+    expect(result).to.deep.equal(item);
+  });
+
+  it('Testing update with returnValues', async () => {
+    expect(await model.upsert(item)).to.deep.equal({ created: true });
+    const result = await model.update({
+      ...item,
+      age: 55
+    }, { returnValues: 'all_old' });
+    expect(result).to.deep.equal(item);
+  });
+
+  it('Testing update with ConditionalCheckFailedException', async ({ capture }) => {
+    expect(await model.upsert(item)).to.deep.equal({ created: true });
+    item.age = 55;
+    const error = await capture(() => model.update(item, { conditions: { attr: 'age', eq: 10 } }));
+    expect(error.code).to.equal('ConditionalCheckFailedException');
+  });
+
+  it('Testing query', async () => {
+    expect(await model.upsert(item)).to.deep.equal({ created: true });
+    const result = await model.query(primaryKey);
+    expect(result).to.deep.equal({
+      payload: [item],
+      page: {
+        next: null,
+        index: { current: 1 },
+        size: 20
+      }
+    });
+  });
+
+  it('Testing query with limit', async () => {
+    expect(await model.upsert(item)).to.deep.equal({ created: true });
+    const result = await model.query(primaryKey, { limit: 1 });
+    expect(result).to.deep.equal({
+      payload: [item],
+      page: {
+        next: {
+          limit: 1,
+          // eslint-disable-next-line max-len
+          cursor: 'eyJsaW1pdCI6MSwic2NhbkluZGV4Rm9yd2FyZCI6dHJ1ZSwibGFzdEV2YWx1YXRlZEtleSI6eyJuYW1lIjoibmFtZSIsImlkIjoiMTIzIn0sImN1cnJlbnRQYWdlIjoyfQ=='
+        },
+        index: { current: 1 },
+        size: 1
+      }
+    });
+  });
+
+  it('Testing query with toReturn', async () => {
+    expect(await model.upsert(item)).to.deep.equal({ created: true });
+    const result = await model.query(primaryKey, { toReturn: ['name'] });
+    expect(result).to.deep.equal({
+      payload: [{ name: 'name' }],
+      page: {
+        next: null,
+        index: { current: 1 },
+        size: 20
+      }
+    });
+  });
+
+  it('Testing query with index', async () => {
+    expect(await model.upsert(item)).to.deep.equal({ created: true });
+    const result = await model.query(primaryKey, {
+      index: 'targetIndex',
+      consistent: false
+    });
+    expect(result).to.deep.equal({
+      payload: [item],
+      page: {
+        next: null,
+        index: { current: 1 },
+        size: 20
+      }
+    });
+  });
+
+  it('Testing query with cursor', async () => {
+    expect(await model.upsert(item)).to.deep.equal({ created: true });
```

Comment:
add two and use first cursor. Don't magically use cursor, genereate cursor and then use it